src/python/tests/test_orthology_benchmark.py

```diff
@@ -515,44 +515,37 @@ def test_get_neighbours(gene: str, n: int, exp_genes: List[Tuple[str, str, float
          [["3", "31", 1, "+"], ["3", "32", 101, "+"], ["3", "33", 201, "+"], ["3", "34", 301, "+"],
           ["3", "35", 401, "+"], ["3", "36", 501, "+"], ["3", "37", 601, "+"], ["3", "38", 701, "+"],
           ["3", "39", 801, "+"]],
-         2, 4, 100),
+         2, 100),
         # All neighbours have an orthologous match and corresponding strands are in agreement
         ("24", "32",
          [["2", "23", 201, "+"], ["2", "24", 301, "+"], ["2", "25", 401, "+"]],
          [["3", "31", 1, "+"], ["3", "32", 101, "+"], ["3", "33", 201, "+"]],
-         1, 0, 100),
+         1, 100),
         # Genes of interest on the opposite strands and all neighbours have an orthologous match on the
         # opposite strand
         ("42", "62",
          [["4", "41", 1, "-"], ["4", "42", 101, "-"], ["4", "43", 201, "+"]],
          [["6", "61", 1, "-"], ["6", "62", 101, "+"], ["6", "63", 201, "+"]],
-         1, 2, 100),
-        # Genes of interest on the opposite strands and one neighbour has an orthologous match on the same
+         1, 100),
+        # Genes of interest on the opposite strands and neighbours have an orthologous match on the same
         # strand
         ("45", "52",
          [["4", "43", 201, "+"], ["4", "44", 301, "+"], ["4", "45", 401, "+"], ["4", "46", 501, "-"],
           ["4", "47", 601, "+"]],
          [["5", "51", 1, "-"], ["5", "52", 101, "-"], ["5", "53", 201, "-"], ["5", "54", 301, "+"]],
-         2, 0, 25),
-        # Too large gap between orthologous matches
-        ("72", "82",
-         [["7", "71", 1, "+"], ["7", "72", 101, "+"], ["7", "73", 201, "+"]],
-         [["8", "81", 1, "+"], ["8", "82", 101, "+"], ["8", "83", 201, "+"], ["8", "84", 301, "+"],
-          ["8", "85", 401, "+"], ["8", "86", 501, "+"], ["8", "87", 601, "+"]],
-         2, 3, 0),
-        # An orthologous match preceding the last (current) match, hence, should not be taken into account
-        # (and a collapsed paralogous neighbour ("26"))
+         2, 25),
+        # Orthologous matches don't appear sequentially among the `neighbours2` (and a collapsed paralogous
+        # neighbour ("26"))
         ("36", "22",
          [["3", "34", 301, "+"], ["3", "35", 401, "+"], ["3", "36", 501, "+"], ["3", "37", 601, "+"],
           ["3", "38", 701, "+"]],
          [["2", "21", 1, "+"], ["2", "22", 101, "+"], ["2", "23", 201, "+"], ["2", "24", 301, "+"],
           ["2", "25", 401, "+"], ["2", "27", 601, "+"], ["2", "28", 701, "+"], ["2", "29", 801, "+"]],
-         2, 4, 50)
+         2, 100)
     ]
 )
 def test_calculate_goc_genes(gene1: str, gene2: str, neighbours1: List[Tuple[str, str, int, str]],
```

Comment:
Does the test only assume valid inputs (with valid data and invalid data)?  If affirmative, should be interesting to include invalid inputs as well (i.e. empty data frame or dataframe with negative values)?

Reply:
Well spotted 🙂 As I said above in one of my comments, I will address this. Thank you!